lib/run-context.js

```diff
@@ -209,6 +212,20 @@ RunContext.prototype.inTmpDir = function(cb) {
   return this.inDir(tmpdir, cb);
 };
 
+/**
+ * Create an environment
+ *
+ * This method is called automatically when creating a RunContext. Only use it if you need
+ * to use the callback.
+ *
+ * @param {Function} [cb] - callback who'll receive the folder path as argument
+ * @return {this} run context instance
+ */
+RunContext.prototype.withEnvironment = function(cb = env => env) {
```

Comment:
Let's set the default in the constructor.

I think it's fair to expect `withEnvironment` to be called with a callback argument; I don't think there's a good reason to provide a default argument.

Making this change should simplify the type check done here `this.env = (this.envCB || (env => env))` (`this.envCB` would always be defined)

Reply:
Done